src/workerd/api/global-scope.c++

```diff
@@ -514,21 +531,25 @@ TimeoutId::NumberType ServiceWorkerGlobalScope::setTimeout(
     v8::Isolate* isolate) {
   auto argv = kj::heapArrayFromIterable<jsg::Value>(kj::mv(args));
   auto timeoutId = IoContext::current().setTimeoutImpl(
-    timeoutIdGenerator,
-    /* repeats = */ false,
-    [function = function.addRef(isolate), argv = kj::mv(argv)](jsg::Lock& js) mutable {
-      auto isolate = js.v8Isolate;
-      auto context = isolate->GetCurrentContext();
-      auto localFunction = function.getHandle(isolate);
-      auto localArgs = KJ_MAP(arg, argv) {
-        return arg.getHandle(isolate);
-      };
-      auto argc = localArgs.size();
+      timeoutIdGenerator,
+      /* repeats = */ false,
+      [function = function.addRef(isolate),
```

Comment:
Optional: Arguably you could do `function = AsyncContextFrame::current()->wrap(function)` at the top of `setTimeout()` and then you don't have to modify the rest of the code. Slightly simpler code.